components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -175,6 +178,15 @@ IJsonSerializer jsonSerializer
                         UpdatedJob = null,
                     };
                 }
+                if (info != null) {
+                    foreach (var assignedJob in _assignedJobs) {
+                    
+                        if (info.Id == assignedJob.Value.Job.Id) {
+                            _publisherDiagnosticInfo.AddOrUpdate(assignedJob.Value.Job.Id, info);
```

Comment:
>_publisherDiagnosticInfo [](http://example.com/codeflow?start=28&length=24)

I think we need to handle also the case when jobs are deleted from the configuration. I'm not sure if when this happen, the diagnostic info map is updated to reflect the deletion of the job.

Reply:
Daniele, please also add tests that validate that no diagnostic info will be reported after an endpoint (data set) is removed.

Reply:
This is handled in RefreshJobs. 
when there is a change _publisherDiagnosticInfo is re-initialized and then in SendHeartbeatAsync is set with the assigned jobs.
I will add and e2e test for this case

Reply:
@dacolgit, we should also perform cleanup of `_publisherDiagnosticInfo` whenever `_assignedJobs` is cleaned. Please look for places where we have `_assignedJobs.Clear()` and add cleanup of `_publisherDiagnosticInfo` just after that. There should be three instances like this.

Reply:
added